src/control/server/commands.go

```diff
@@ -149,3 +150,68 @@ func (p *PrepNvmeCmd) Execute(args []string) error {
 	// never reached
 	return nil
 }
+
+// PrepScmCmd is the struct representing the command to prep SCM modules by
+// configuring in AppDirect mode and creating relevant namespaces.
+type PrepScmCmd struct {
+	Reset bool `short:"r" long:"reset" description:"Reset modules to memory mode after removing namespaces"`
+}
+
+// Execute is run when PrepScmCmd activates
+//
+// Perform task then exit immediately. No config parsing performed.
+func (p *PrepScmCmd) Execute(args []string) error {
+	ok, _ := common.CheckSudo()
+	if !ok {
+		return errors.New("subcommand must be run as root or sudo")
+	}
+
+	config := newConfiguration()
+
+	server, err := newControlService(
+		&config, getDrpcClientConnection(config.SocketDir))
+	if err != nil {
+		return errors.WithMessage(err, "initialising ControlService")
+	}
+
+	fmt.Println("Scanning locally-attached SCM storage...")
+	resp := new(pb.ScanStorageResp)
+	server.scm.Discover(resp)
+
+	if resp.Scmstate.Status != pb.ResponseStatus_CTRL_SUCCESS {
+		fmt.Fprintln(os.Stderr, "scm scan: "+resp.Scmstate.Error)
+		os.Exit(1)
+	}
+
+	if server.scm.initialized {
```

Comment:
This is a little confusing... If the initialized variable is set, return a "not initialized" error? Is the variable misnamed or is there a bug here?

Reply:
yes fixed